### PR TITLE
Consolidate waking up code

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
SUMMARY: None

#### Purpose of change
Almost the same code was present in `avatar::wake_up` and `player_hardcoded_effects`.  In particular, this code duplication meant that I missed some wake up cases when I added the event for waking up, making the survival achievement fail to trigger sometimes.

#### Describe the solution
Consolidate the two into one implementation.

#### Describe alternatives you've considered
The two were very slightly different.  Based on my analysis of the code I think my merged implementation is equivalent or better, but it's hard to think about (or test) every corner case.  Waking up interacts with alarms in a complicated way.

#### Testing
I played the game and ensured that waking normally and via an alarm seemed to work as expected.

#### Additional context
Discovered while testing achievements.

### CODE REVIEW

File: src/player_hardcoded_effects.cpp [hunk 1]
```diff
@@ -1235,25 +1234,10 @@ void player::hardcoded_effects( effect &it )
...
-                add_msg_if_player( _( "It looks like you woke up just before your alarm." ) );
-                remove_effect( effect_alarm_clock );
-            } else if( has_effect( effect_slept_through_alarm ) ) { // slept though the alarm.
-                if( has_bionic( bio_watch ) ) {
-                    add_msg_if_player( m_warning, _( "It looks like you've slept through your internal alarm…" ) );
-                } else {
-                    add_msg_if_player( m_warning, _( "It looks like you've slept through the alarm…" ) );
-                }
-                get_effect( effect_slept_through_alarm ).set_duration( 0_turns );
-                remove_effect( effect_alarm_clock );
```
Comment (ifreund) [inline src/player_hardcoded_effects.cpp:1255]: I don't see these two lines present after the change. Is that intentional?

Comment (jbytheway) [reply src/player_hardcoded_effects.cpp:1255]: `avatar::wake_up` calls `Character::wake_up`, and that removes these two effects (albeit with slightly different code).
